Mathlib/LinearAlgebra/PerfectPairing.lean

```diff
@@ -57,7 +59,63 @@ protected def flip : PerfectPairing R N M where
 
 @[simp] lemma flip_flip : p.flip.flip = p := rfl
 
--- TODO `M` and `N` are both reflexive
+/-- The linear equivalence from `M` to `Dual R N` induced by a perfect pairing. -/
+noncomputable def toDualLeft : M ≃ₗ[R] Dual R N :=
+  LinearEquiv.ofBijective p.toLin p.bijectiveLeft
+
+@[simp]
+theorem toDualLeft_apply (a : M) : p.toDualLeft a = p.toLin a :=
+  rfl
+
+@[simp]
+theorem apply_toDualLeft_symm_apply (f : Dual R N) (x : N) : p (p.toDualLeft.symm f) x = f x := by
+  have h := LinearEquiv.apply_symm_apply p.toDualLeft f
+  rw [toDualLeft_apply] at h
+  exact congrFun (congrArg DFunLike.coe h) x
+
+/-- The linear equivalence from `N` to `Dual R M` induced by a perfect pairing. -/
+noncomputable def toDualRight : N ≃ₗ[R] Dual R M :=
+  toDualLeft p.flip
+
+@[simp]
+theorem toDualRight_apply (a : N) : p.toDualRight a = p.flip.toLin a :=
+  rfl
+
+@[simp]
+theorem apply_apply_toDualRight_symm (x : M) (f : Dual R M) :
+    (p x) (p.toDualRight.symm f) = f x := by
+  have h := LinearEquiv.apply_symm_apply p.toDualRight f
+  rw [toDualRight_apply] at h
+  exact congrFun (congrArg DFunLike.coe h) x
+
+theorem toDualLeft_of_toDualRight_symm (x : M) (f : Dual R M) :
+    (p.toDualLeft x) (p.toDualRight.symm f) = f x := by
+  rw [@toDualLeft_apply]
+  exact apply_apply_toDualRight_symm p x f
+
+theorem toDualRight_symm_toDualLeft (x : M) :
+    p.toDualRight.symm.dualMap (p.toDualLeft x) = Dual.eval R M x := by
+  ext f
+  simp only [LinearEquiv.dualMap_apply, Dual.eval_apply]
+  exact toDualLeft_of_toDualRight_symm p x f
+
```

Comment:
If you have a lemma `p.toDualRight.symm.dualMap.comp p.toDualLeft = Dual.eval R M`, the proof of  `reflexive_left` can be reduced to `rw [<- this]; exact p.bijective_toDualRight_symm_toDualLeft`.

Reply:
I had tried it earlier, but had given up in the face of a `typeclass instance problem is stuck` error.  Thank you for the extra push.